myhoard/controller.py

```diff
@@ -1163,13 +1174,23 @@ def _purge_old_binlogs(self, *, mysql_maybe_not_running=False):
                 last_purge = time.time()
                 last_could_have_purged = last_purge
         finally:
+            current_time = time.time()
+            
             self.state_manager.update_state(
-                binlogs_purged_at=time.time(),
+                binlogs_purged_at=current_time,
                 last_binlog_purge=last_purge,
                 last_could_have_purged=last_could_have_purged,
             )
-            self.stats.gauge_float("myhoard.binlog.time_since_any_purged", time.time() - last_purge)
-            self.stats.gauge_float("myhoard.binlog.time_since_could_have_purged", time.time() - last_could_have_purged)
+
+            self.stats.gauge_float("myhoard.binlog.time_since_any_purged", current_time - last_purge)
+            self.stats.gauge_float("myhoard.binlog.time_since_could_have_purged", current_time - last_could_have_purged)
+
+            oldest_binlog_time = self._get_oldest_binlog_time()
+            self.stats.gauge_float(
+                "myhoard.binlog.time_since_oldest_should_have_purged", 
+                current_time - (oldest_binlog_time + purge_settings["min_binlog_age_before_purge"]),
+                tags = { "support_oldest_should_have_purged" : 1 }
```

Comment:
I meant adding this tag to the existing time_since_could_have_purged metric so that it is easy to not use old alert for any metric points that do have the tag present. Also would be clearer to pass the value directly as a string since tags are always strings and either it gets converted into one later or then there will be some error.

Reply:
Ah ok made this change